jobqueue/jobqueue.go

```diff
@@ -92,6 +92,10 @@ func (q *jobQueue) Pop(limit uint) ([]Job, error) {
 
 func (q *jobQueue) Complete(job Job, res *Result) {
 	j := &completedJob{job, 0}
+	if res.IsFailure() && !j.canRetry() {
+		res.Status = ResultStatusPermanentFailure
```

Comment:
I think overwriting `res.Status` isn't a good idea because we won't be able to distinguish a permanent failure (returned by a worker) from the retry limit being exceeded, especially in the `failureLog`.  Shouldn't we make a copy of `res` here and use it for condition but use the original `res` for logging?